### PR TITLE
Add dockerfile and docker make targets

### PR DESCRIPTION
**Pull request by squidboylan**

### CODE REVIEW

File: Makefile [hunk 1]
```diff
@@ -6,12 +6,21 @@ OBJ := start.o \
 
 CFLAGS := -O0 -ffreestanding -nostdlib -Wall -Wextra -g
 
+# This lets us override CC and LD with environment variables
```
Comment (connorkuehl) [inline Makefile]: Aren't `CC` and `LD` already environment variables? If they're set in the shell session before invoking `make` won't the correct ones be invoked?

File: Makefile [hunk 2]
```diff
@@ -6,12 +6,21 @@ OBJ := start.o \
...
+	LD := $(LD)
+endif
+
 all: $(KERNBIN)
 
 $(KERNBIN): $(OBJ)
 	$(LD) -T $(LDSCRIPT) -o $@ $^
 
-.PHONY: clean qemu qemu-gdb
+.PHONY: clean docker-image docker-qemu docker-qemu qemu qemu-gdb
```
Comment (connorkuehl) [inline Makefile]: I think you put an extra `docker-qemu` here.

Is it possible to add a `docker-qemu-gdb` target?

Comment (squidboylan) [reply Makefile]: thanks! Yeah probably, i need to think about the best way to do that but i should be able to

Comment (squidboylan) [reply Makefile]: oh you mean to run qemu with the right flags to later attach with gdb and not have a make target for starting gdb. I can add both of those i think

Comment (connorkuehl) [reply Makefile]: Yeah, running `gdb` can be manual, but a target for telling `qemu` to expose a debug port and to wait for the debugger to attach is what I'm after.

File: Makefile (hunk 1 above)
Comment (squidboylan) [reply Makefile]: You're right, thanks

**Review by connorkuehl — APPROVED** [review]

**Pull request merged**